### PR TITLE
Fix bug in content function index expansion

### PR DESCRIPTION
**Pull request by apmoriarty**
Fix bug in content function index expansion when a phrase is a repeated term. Addresses #1484

### CODE REVIEW

**Comment by apmoriarty** [comment]
@FineAndDandy @hlgp @jwomeara If you all could provide a review that would be much appreciated

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java [hunk 1]
```diff
@@ -461,15 +464,19 @@ public static JexlNode distributeFunctionIntoIndexQuery(JexlNode function, JexlN
...
          */
         public static JexlNode distributeFunctionIntoComponent(JexlNode function, JexlNode component) {
             List<JexlNode> children = new LinkedList<>();
             String field = findCommonField(component);
             if (field != null) {
                 children.add(updateFunctionWithField(function, field));
-                children.addAll(Arrays.asList(JexlNodes.children(component)));
+                if (component instanceof ASTEQNode) {
+                    children.add(component);
+                } else {
```
Comment (FineAndDandy) [inline warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java]: `else if { ASTAndNode } else { //warn|error}` ?

**Review by FineAndDandy — CHANGES_REQUESTED** [review]

**Comment by jwomeara** [comment]
What issue is this tied to?

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java (hunk 1 above)
Comment (jwomeara) [reply warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java]: i agree

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java [hunk 2]
```diff
@@ -424,9 +424,12 @@ public static JexlNode distributeFunctionIntoIndexQuery(JexlNode function, JexlN
             List<JexlNode> components = new LinkedList<>();
             
             JexlNode deref = JexlASTHelper.dereference(indexQuery);
-            if (deref instanceof ASTAndNode) {
-                // single component
+            if (deref instanceof ASTAndNode || deref instanceof ASTEQNode) {
+                
+                // it is possible that an index query contains only a single equality node
+                // i.e., "phrase('foo', 'foo')" would produce a single equality node "FOO == 'bar'"
```
Comment (jwomeara) [inline warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java]: Is this comment correct?  `phrase(foo, foo)` becomes `FOO == 'bar'` ?

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java]: Added in next commit

**Comment by apmoriarty** [comment]
> What issue is this tied to?

Wrote up issue https://github.com/NationalSecurityAgency/datawave/issues/1484

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java]: I'll make this language more precise. The index expansion for that function will produce a single equality node.

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java [hunk 3]
```diff
@@ -424,9 +424,12 @@ public static JexlNode distributeFunctionIntoIndexQuery(JexlNode function, JexlN
...
                 components.add(deref);
+                
             } else if (deref instanceof ASTOrNode) {
                 // multiple components
                 for (JexlNode child : JexlNodes.children(deref)) {
```
Comment (jwomeara) [inline warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java:435]: On line 439 theres an exception whose message should be updated to include EQ nodes.

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java (hunk 2 above)
Comment (jwomeara) [reply warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java]: my question was more about it going from foo, foo, to foo, bar.  is that right?

**Review by jwomeara — CHANGES_REQUESTED** [review]
See comments.

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java]: As written, that is incorrect. I updated the comment to be " `phrase(termOffsetMap,'bar','bar'` produces index query `"FOO == 'bar'"` "

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java (hunk 3 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java:435]: ahh, good catch

**Review by jwomeara — APPROVED** [review]

**Review by FineAndDandy — APPROVED** [review]

**Pull request merged**